### PR TITLE
" Download Vector Tiles To Local Cache" new Sample

### PR DESCRIPTION
**Pull request by changanxian**
#1561

Implements the new sample "Download vector tiles to local cache" following the [common design](https://devtopia.esri.com/runtime/common-samples/tree/main/designs/DownloadVectorTilesToLocalCache)

- Notes for the reviewer: the 'generated_samples_list.json' is not updated for the changes made in the "set_basemap" and "show_device_location_history", so they are reflected in the PR.

### CODE REVIEW

File: lib/samples/download_vector_tiles_to_local_cache/download_vector_tiles_to_local_cache_sample.dart [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+              _exportVectorTilesJob?.result as ExportVectorTilesResult;
+          _loadExportedVectorTiles(result);
+          setState(() => _previewMap = true);
+          setState(() => _isJobStarted = false);
+        }
+      },
+    );
+
+    // Start the export vector tiles job.
+    _exportVectorTilesJob?.start();
```
Comment (PaulAllanSturm) [inline lib/samples/download_vector_tiles_to_local_cache/download_vector_tiles_to_local_cache_sample.dart:267]: Normally for Jobs, we don't use `start()` and `onStatusChanged()`. Normally, we would structure it like this:
```dart
try {
  final result = await _exportVectorTilesJob?.run();
  // code to process result ...
} on ArcGISException catch (e) {
  showDialog(
    // code for an error dialog ...
 );
}
setState(() => _isJobStarted = false);
```

It's not wrong to use `start()` and `onStatusChanged()`, but I think it's clearer to use `run()` in a `try/catch` block.

**Comment by changanxian** [comment]
@PaulAllanSturm  your comments have been addressed, please review

File: lib/samples/download_vector_tiles_to_local_cache/download_vector_tiles_to_local_cache_sample.dart [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+    // Create the export vector tiles job.
+    _exportVectorTilesJob =
+        vectorTilesExportTask.exportVectorTilesWithItemResourceCache(
+      parameters: exportVectorTilesParameters,
+      vectorTileCacheUri: vtpkFile.uri,
+      itemResourceCacheUri: Uri.directory(resourceDirectory),
+    );
+
+    // Listen to the job's progress, status, and completion.
```
Comment (PaulAllanSturm) [inline lib/samples/download_vector_tiles_to_local_cache/download_vector_tiles_to_local_cache_sample.dart:201]: The comment should be updated to just `// Listen to the job's progress.`

File: lib/samples/download_vector_tiles_to_local_cache/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+* ExportVectorTilesJob
+* ExportVectorTilesParameters
+* ExportVectorTilesResult
+* ExportVectorTilesTask
+* ItemResourceCache
+* VectorTileCache
+
+## Additional information
+
+**NOTE:** Downloading Tiles for offline use requires authentication with the web map's server. To use this sample, you will require an [ArcGIS Online](https://www.arcgis.com) account.
```
[the remark this answers is not in the document]
Comment (changanxian) [reply lib/samples/download_vector_tiles_to_local_cache/README.md:36]: No, this sample doesn't. I was hesitant to remove it.  We may not need to mention this, but I saw the Swift Sample keep the [comments](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/main/Shared/Samples/Download%20vector%20tiles%20to%20local%20cache)

[the remark this answers is not in the document]
Comment (PaulAllanSturm) [reply lib/samples/download_vector_tiles_to_local_cache/README.md:36]: @jenmerritt - do you know why the README for "Download vector tiles to local cache" has a note about requiring an ArcGIS Online account? As far as I can tell, this sample is no different than any other sample.

[the remark this answers is not in the document]
Comment (changanxian) [reply lib/samples/download_vector_tiles_to_local_cache/README.md:36]: @jenmerritt - I merged the PR now, if you feel the paragraph 'NOTE' is unnecessary, I will remove it from a separate PR.

**Pull request merged**